OMG.Classroom/DataAccessLayer/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,51 @@
+﻿using DataAccessLayer.Entities;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer.Repositories
+{
+    public abstract class BaseRepository<E> where E : BaseEntity
+    {
+        public readonly ClassroomDbContext _context;
```

Comment:
All fields start with underscore => they're protected/private. Since this is base repo, make them protected